### DISCUSSION TITLE
How to assign automerge PRs with status failures

### DISCUSSION THREAD

**Discussion by richardsimko** [renovatebot/renovate#12676] [Request Help] [answered]
The docs state:

> By default, Renovate will not assign reviewers and assignees to an automerge-enabled PR unless it fails status checks.

https://docs.renovatebot.com/configuration-options/#assignautomerge

However my experience is Renovate will not assign any PRs that have automerge. As the first time Renovate runs, when it sets up the branch, it doesn't know the state of the build. So it's fair that at this point it does not assign, since the PR might automerge cleanly. Here's the logs:

> DEBUG: Skipping assignees and reviewers as automerge=true(branch="renovate/eslint-plugin-import-2.x")

However when it later runs a second time and the build is failed I expect it to update the PR with an assignee. It does not however and the logs do not at all mention assignees. Is there any way to get this behavior without having every single PR assigned (This would create a lot of noise as our repos require PRs with 1 approver - which we've solved with renovate-approve).

**Comment by rarkins** [comment, marked as answer]
Can you set up a reproduction repo on github.com with a single dependency to update with automerge plus an intentionally failing github action? I think that should then be a valid reproduction which can be forked and debugged.

**Reply by richardsimko** [reply]
I've created https://github.com/WarOnCancer/renovate-test/ with the same config as we run in our regular repos minus the encrypted token for internal packages. Renovate tried to update the one dependency and created a PR with no assignees.

**Reply by richardsimko** [reply]
Can you access build logs for public projects? If not then just let me know and I can send you the logs from the 2 runs.

**Reply by richardsimko** [reply]
And I don't know if it's relevant but we have branch protection enabled for the repos (Same in this test repo) with the following settings. Normally Renovate will create its branch, wait for builds to pass and then just merge it without a PR.
![image](https://user-images.githubusercontent.com/1245031/141978596-70653c19-68af-4310-8647-8384a260257d.png)

**Reply by rarkins** [reply]
I believe https://github.com/renovatebot/renovate/pull/12705 will fix it

**Reply by richardsimko** [reply]
Thank you! :)

**Reply by richardsimko** [reply]
That fixed it, the PR in my test repo was assigned to me now. Thank you!
